vfs/directory.go

```diff
@@ -63,42 +59,50 @@ func (d *DirDoc) SetRev(rev string) {
 // ToJSONApi implements temporary interface JSONApier to serialize
 // the directory document
 func (d *DirDoc) ToJSONApi() ([]byte, error) {
+	attrs := map[string]interface{}{
+		"name":       d.Name,
+		"created_at": d.CreatedAt,
+		"updated_at": d.UpdatedAt,
+		"tags":       d.Tags,
+	}
 	data := map[string]interface{}{
 		"type":       d.DocType(),
 		"id":         d.ID(),
 		"rev":        d.Rev(),
-		"attributes": d.Attrs,
+		"attributes": attrs,
 	}
 	m := map[string]interface{}{
 		"data": data,
 	}
 	return json.Marshal(m)
 }
 
-// CreateDirectory is the method for creating a new directory
-func CreateDirectory(m *DocAttributes, fs afero.Fs, dbPrefix string) (doc *DirDoc, err error) {
-	if m.docType != FolderDocType {
-		err = ErrDocTypeInvalid
-		return
-	}
-
-	pth, _, err := createNewFilePath(m, fs, dbPrefix)
-	if err != nil {
+// NewDirDoc is the DirDoc constructor. The given name is validated.
+func NewDirDoc(name, folderID string, tags []string) (doc *DirDoc, err error) {
+	if err = checkFileName(name); err != nil {
```

Comment:
Coding style decision to be made here : 
@nono reverted this kind of set-output-value-and-return-empty in one of its PR (although it was only 1 value)

We need to pick a style and stick to it. Personnaly, I am not sure : this is cleaner than `return nil, err` but the intent is less readable. When in doubt, it is better to pick more readability rather than a few less chars.